packages/rocket-webhook-local-infrastructure/src/controllers/webhook-controller.ts

```diff
@@ -1,48 +1,75 @@
-import { functionID } from '@boostercloud/rocket-webhook-types'
-
-export type APIResult =
-  | { status: 'success'; result: unknown; headers: Record<string, number | string | ReadonlyArray<string>> }
-  | { status: 'failure'; code: number; title: string; reason: string }
+import {
+  functionID,
+  Headers,
+  HttpSuccessStatusCode,
+  WebhookAPIResult,
+  WebhookAPISuccessResult,
+  WebhookResponseType,
+} from '@boostercloud/rocket-webhook-types'
+import * as stream from 'stream'
 
 export class WebhookController {
   public router: express.Router = express.Router()
 
-  constructor(origin: string) {
-    this.router.post(`/${origin}`, express.raw({ type: 'application/json' }), this.handleWebhook.bind(this))
+  constructor(readonly endpoint: string) {
+    this.router.post(`/${endpoint}`, express.raw(), this.handleWebhook.bind(this))
   }
 
   public async handleWebhook(req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
     try {
       const request = {
         [rocketFunctionIDEnvVar]: functionID,
-        req: {
-          method: 'POST',
-          url: req.url,
-          originalUrl: req.originalUrl,
-          headers: req.headers,
-          query: req.query,
-          params: req.params,
-          rawBody: req.rawBody,
-          body: req.body,
-        },
+        req: req,
       }
-      const response = (await boosterRocketDispatcher(request)) as APIResult
-      if (response.status === 'success') {
-        for (const headersKey in response.headers) {
-          res.setHeader(headersKey, response.headers[headersKey])
+      const response: WebhookAPIResult = (await boosterRocketDispatcher(request)) as WebhookAPIResult
+      res.status(response.status)
+      if (this.isSuccess(response)) {
+        const body = response.body
+        this.setHeaders(response, res)
+        const type = this.getType(response?.headers)
+        if (type === WebhookResponseType.file) {
+          const readStream = new stream.PassThrough()
+          readStream.end(response.body)
+          readStream.pipe(res)
+        } else if (type === WebhookResponseType.json) {
+          res.json(body)
+        } else {
+          res.send(body)
         }
-        res.status(HttpCodes.Ok).json(response.result)
-      } else {
-        res.status(response.code).json({ title: response.title, reason: response.reason })
+        return
       }
+      res.json(response.body)
     } catch (e) {
       const err = e as Error
       await requestFailed(err, res)
       next(e)
     }
   }
+
+  private getType(headers: Headers): WebhookResponseType {
+    const contentType = headers['Content-type']
+
+    switch (contentType) {
+      case WebhookResponseType.json:
+        return WebhookResponseType.json
+      case WebhookResponseType.text:
+        return WebhookResponseType.text
+      default:
+        return WebhookResponseType.file
+    }
+  }
+
```

Comment:
`headers['Content-type']` could contains anything so this method is returning the same value only if it is known/expected. If the `contentType` is unknown then we return `file`. It could be simplified with:

```typescript
if (contentType && (contentType === WebhookResponseType.json || contentType === WebhookResponseType. text)) {
  return contentType
}
return WebhookResponseType.file
```

but I think it is easy to read with a switch.

Reply:
But, in the place this method is used, we are already contemplating that. This is the code as it is in the PR:
```typescript
        const type = this.getType(response?.headers)
        if (type === WebhookResponseType.file) {
          const readStream = new stream.PassThrough()
          readStream.end(response.body)
          readStream.pipe(res)
        } else if (type === WebhookResponseType.json) {
          res.json(body)
        } else {
          res.send(body)
        }
```
As you can see, we only deal with the content types we care about. We can avoid calling the `this.getType` method and the result would be the same: 
```typescript
        const type = response?.headers['Content-type']
        if (type === WebhookResponseType.file) {
          const readStream = new stream.PassThrough()
          readStream.end(response.body)
          readStream.pipe(res)
        } else if (type === WebhookResponseType.json) {
          res.json(body)
        } else {
          res.send(body)
        }
```

Here we are also only handling the content types we want and, for the rest, just calling the default `res.send()`

Reply:
Ok, this is another thing. Let me fix it